assets/js/leaves.js

```diff
@@ -38,6 +38,9 @@ function cointoss() {
   return Math.random() < 0.5;
 }
 
+const TagSelectEvent = new Event("tag-select");
+const TagDeselectEvent = new Event("tag-deselect");
```

Comment:
And we need these because we want to tag selection to do more than just updateSelection (also zoom, etc)?

Reply:
Yes. There are now zoom behaviors associated with tag interactions. Signals seemed the cleanest way to handle that. Selecting a tag will trigger a zoom out so you can see all the tags with that leaf; deselecting a tag will zoom back in of there is a leaf selected or zoom out if not.